include/libtorrent/aux_/typed_span.hpp

```diff
@@ -49,6 +49,9 @@ namespace libtorrent { namespace aux {
 
 		// pull in constructors from base class
 		using base::base;
+		// implicit constructor needed for template based member functions
+		typed_span() = default;
+		typed_span(span<T> const& v) : base(v) {} // NOLINT
```

Comment:
The problem with this (afaict) is that the ``typed_span`` now has implicit conversions in both directions to ``span`` and also from ``span``. I believe this defeats some of the purpose of having the extra type in the first place. My first attempt at fixing this was to make this constructor private, since (I believe) this implicit conversion is only necessary from within the member functions themselves. Would that work?